### PR TITLE
chore(medusa,core-flows): begin and cancel receive return

### PR DESCRIPTION
**Pull request by carlos-r-l-rodrigues**
What:
* `POST /admin/returns/:id/receive` 
* `DELETE /admin/returns/:id/receive`

CLOSES: CC-189

### CODE REVIEW

File: packages/core/core-flows/src/order/workflows/return/begin-receive-return.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        return {
+          change_type: "return" as const,
+          order_id: order.id,
+          return_id: orderReturn.id,
+          created_by: input.created_by,
+          description: input.description,
+          internal_note: input.internal_note,
+        }
+      }
+    )
```
Comment (olivermrbl) [inline packages/core/core-flows/src/order/workflows/return/begin-receive-return.ts:70]: question: Can we tell if this change is for receiving the return? It doesn't look different from the begin return request action, but maybe this is intentional, so just asking out of curiosity.

File: packages/core/core-flows/src/order/workflows/return/cancel-receive-return.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    }).config({ name: "order-query" })
+
+    const orderChange: OrderChangeDTO = useRemoteQueryStep({
+      entry_point: "order_change",
+      fields: ["id", "status", "version"],
+      variables: {
+        filters: {
+          order_id: orderReturn.order_id,
+          return_id: orderReturn.id,
+          status: [OrderChangeStatus.PENDING, OrderChangeStatus.REQUESTED],
```
Comment (olivermrbl) [inline packages/core/core-flows/src/order/workflows/return/cancel-receive-return.ts:59]: question: Can a receivable return be pending?

File: packages/medusa/src/api/admin/returns/[id]/receive/route.ts [hunk 3]
```diff
@@ -9,12 +16,52 @@ export const POST = async (
...
+      },
+    },
+    fields: req.remoteQueryConfig.fields,
   })
 
-  res.status(200).json({ return: result })
+  const [orderReturn] = await remoteQuery(queryObject)
+
+  res.json({
+    order_change: result,
```
Comment (olivermrbl) [inline packages/medusa/src/api/admin/returns/[id]/receive/route.ts]: thought: Would it make sense to return the order + return?

**Review by olivermrbl — APPROVED** [review]
Overall LGTM, just had a few questions 💪

File: packages/core/core-flows/src/order/workflows/return/begin-receive-return.ts (hunk 1 above)
Comment (carlos-r-l-rodrigues) [reply packages/core/core-flows/src/order/workflows/return/begin-receive-return.ts:70]: This type is generic and it is just covering "return", "claim", "exchange" and "edit". It is indeed the same as begin return request, the difference is that the return already exist.

File: packages/core/core-flows/src/order/workflows/return/cancel-receive-return.ts (hunk 2 above)
Comment (carlos-r-l-rodrigues) [reply packages/core/core-flows/src/order/workflows/return/cancel-receive-return.ts:59]: This is only the status of the OrderChange entity. Not the return itself.
This status is pending while the changes are being done in the UI and they are not yet confirmed.

File: packages/medusa/src/api/admin/returns/[id]/receive/route.ts (hunk 3 above)
Comment (carlos-r-l-rodrigues) [reply packages/medusa/src/api/admin/returns/[id]/receive/route.ts]: yeah, we can do it.
I will change it

**Pull request merged**